hnn_core/tests/test_network.py

```diff
@@ -1911,3 +1923,152 @@ def test_update_weights_metadata():
+    custom_gain = 2.5
+    net.add_connection(
+        src_gids="L5_pyramidal",
+        target_gids="L5_basket",
+        loc="soma",
+        receptor="ampa",
+        weight=1e-3,
+        delay=1.0,
+        lamtha=3.0,
+        gain=custom_gain,
+    )
+
+    # Check that the gain was set correctly
+    conn_idx = pick_connection(
+        net, src_gids="L5_pyramidal", target_gids="L5_basket", receptor="ampa"
+    )[-1]
+    assert net.connectivity[conn_idx]["nc_dict"]["gain"] == custom_gain
+
+    # Add connection with both custom threshold and gain
+    net.add_connection(
+        src_gids="L2_basket",
+        target_gids="L2_pyramidal",
+        loc="soma",
+        receptor="gabaa",
+        weight=1e-3,
+        delay=1.0,
+        lamtha=3.0,
+        threshold=custom_threshold,
+        gain=custom_gain,
+    )
+
+    # Check that both were set correctly
+    conn_idx = pick_connection(
+        net, src_gids="L2_basket", target_gids="L2_pyramidal", receptor="gabaa"
+    )[-1]
+    assert net.connectivity[conn_idx]["nc_dict"]["threshold"] == custom_threshold
+    assert net.connectivity[conn_idx]["nc_dict"]["gain"] == custom_gain
+
+    # Test that default threshold is inherited from network when threshold=None
+    net.add_connection(
+        src_gids="L5_basket",
+        target_gids="L5_pyramidal",
+        loc="soma",
+        receptor="gabaa",
+        weight=1e-3,
+        delay=1.0,
+        lamtha=3.0,
+        threshold=None,  # Should use net.threshold
+        gain=1.5,
+    )
+
+    conn_idx = pick_connection(
+        net, src_gids="L5_basket", target_gids="L5_pyramidal", receptor="gabaa"
+    )[-1]
+    assert net.connectivity[conn_idx]["nc_dict"]["threshold"] == net.threshold
+    assert net.connectivity[conn_idx]["nc_dict"]["gain"] == 1.5
+
+
+def test_get_cell_index_by_synapse_type():
+    """Test _get_cell_index_by_synapse_type helper function."""
+    net = jones_2009_model()
+
+    e_gids, i_gids = _get_cell_index_by_synapse_type(net)
+
+    # Check that excitatory cells are correctly identified
+    expected_e_gids = list(net.gid_ranges["L2_pyramidal"]) + list(
+        net.gid_ranges["L5_pyramidal"]
+    )
+    assert sorted(e_gids) == sorted(expected_e_gids)
+
+    # Check that inhibitory cells are correctly identified
+    expected_i_gids = list(net.gid_ranges["L2_basket"]) + list(
+        net.gid_ranges["L5_basket"]
+    )
+    assert sorted(i_gids) == sorted(expected_i_gids)
+
+    # Ensure no overlap between excitatory and inhibitory
+    assert len(set(e_gids).intersection(set(i_gids))) == 0
+
+
+def test_check_global_synaptic_gains_uniformity():
+    """Test _check_global_synaptic_gains_uniformity function."""
+    net = jones_2009_model()
+
+    # Set some global gains
+    net.set_global_synaptic_gains(e_e=0.5)
+
+    # With uniform gains (default), should return False (no warning)
+    result = _check_global_synaptic_gains_uniformity(net)
+    assert result is False
```

Comment:
To me the title of this function implies that `result` should be true when global synaptic gains are uniform